hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTask.java

```diff
@@ -205,15 +218,38 @@ public Pair<String, Boolean> reprocess(OMMetadataManager omMetadataManager) {
       return new ImmutablePair<>(getTaskName(), false);
     }
 
+    try {
+      writeNSSummariesToDB(nsSummaryMap);
+    } catch (IOException e) {
+      LOG.error("Unable to write Namespace Summary data in Recon DB.", e);
+      return new ImmutablePair<>(getTaskName(), false);
+    }
     LOG.info("Completed a reprocess run of NSSummaryTask");
     return new ImmutablePair<>(getTaskName(), true);
   }
 
-  private void writeOmKeyInfoOnNamespaceDB(OmKeyInfo keyInfo)
-          throws IOException {
+  private void writeNSSummariesToDB(Map<Long, NSSummary> nsSummaryMap)
+      throws IOException {
+    RDBBatchOperation rdbBatchOperation = new RDBBatchOperation();
+    nsSummaryMap.keySet().forEach((Long key) -> {
+      try {
+        reconNamespaceSummaryManager.batchStoreNSSummaries(rdbBatchOperation,
+            key, nsSummaryMap.get(key));
+      } catch (IOException e) {
+        LOG.error("Unable to write Namespace Summary data in Recon DB.",
+            e);
+      }
+    });
+    reconNamespaceSummaryManager.commitBatchOperation(rdbBatchOperation);
+  }
+
+  private void handlePutKeyEvent(OmKeyInfo keyInfo, Map<Long,
+      NSSummary> nsSummaryMap) throws IOException {
     long parentObjectId = keyInfo.getParentObjectID();
-    NSSummary nsSummary = reconNamespaceSummaryManager
-            .getNSSummary(parentObjectId);
+    NSSummary nsSummary = nsSummaryMap.get(parentObjectId);
+    if (nsSummary == null) {
+      nsSummary = reconNamespaceSummaryManager.getNSSummary(parentObjectId);
```

Comment:
If `nsSummaryMap.get()` is not expected to return null, we can add a `LOG.warn` or `LOG.error` here. If `nsSummaryMap.get()` could return null, we can add a `LOG.debug` message instead.

Maybe adding some comments would also help others understand when would `nsSummary` be null in the first place.

Reply:
Got it. Thx!